service/frontend/wrappers/ratelimited/api_generated.go

```diff
@@ -0,0 +1,417 @@
+	if err != nil {
+		return
+	}
+	domainName, err := h.domainCache.GetDomainName(token.DomainID)
+	if err != nil {
+		return
+	}
+	// Count the request in the host RPS,
+	// but we still accept it even if RPS is exceeded
+	h.allowDomain(ratelimitTypeWorker, domainName)
+	return h.wrapped.RespondActivityTaskFailed(ctx, rp1)
+}
+
+func (h *apiHandler) RespondActivityTaskFailedByID(ctx context.Context, rp1 *types.RespondActivityTaskFailedByIDRequest) (err error) {
+	// Count the request in the host RPS,
+	// but we still accept it even if RPS is exceeded
+	h.allowDomain(ratelimitTypeWorker, rp1.GetDomain())
+	return h.wrapped.RespondActivityTaskFailedByID(ctx, rp1)
+}
+
+func (h *apiHandler) RespondDecisionTaskCompleted(ctx context.Context, rp1 *types.RespondDecisionTaskCompletedRequest) (rp2 *types.RespondDecisionTaskCompletedResponse, err error) {
+	token, err := h.tokenSerializer.Deserialize(rp1.TaskToken)
+	if err != nil {
+		return
+	}
+	domainName, err := h.domainCache.GetDomainName(token.DomainID)
+	if err != nil {
+		return
+	}
+	// Count the request in the host RPS,
+	// but we still accept it even if RPS is exceeded
+	h.allowDomain(ratelimitTypeWorker, domainName)
+	return h.wrapped.RespondDecisionTaskCompleted(ctx, rp1)
+}
+
+func (h *apiHandler) RespondDecisionTaskFailed(ctx context.Context, rp1 *types.RespondDecisionTaskFailedRequest) (err error) {
+	token, err := h.tokenSerializer.Deserialize(rp1.TaskToken)
+	if err != nil {
+		return
+	}
+	domainName, err := h.domainCache.GetDomainName(token.DomainID)
+	if err != nil {
+		return
+	}
+	// Count the request in the host RPS,
+	// but we still accept it even if RPS is exceeded
+	h.allowDomain(ratelimitTypeWorker, domainName)
+	return h.wrapped.RespondDecisionTaskFailed(ctx, rp1)
+}
+
+func (h *apiHandler) RespondQueryTaskCompleted(ctx context.Context, rp1 *types.RespondQueryTaskCompletedRequest) (err error) {
+	token, err := h.tokenSerializer.DeserializeQueryTaskToken(rp1.TaskToken)
+	if err != nil {
+		return
+	}
+	domainName, err := h.domainCache.GetDomainName(token.DomainID)
+	if err != nil {
+		return
+	}
+	// Count the request in the host RPS,
+	// but we still accept it even if RPS is exceeded
+	h.allowDomain(ratelimitTypeWorker, domainName)
+	return h.wrapped.RespondQueryTaskCompleted(ctx, rp1)
+}
+
+func (h *apiHandler) RestartWorkflowExecution(ctx context.Context, rp1 *types.RestartWorkflowExecutionRequest) (rp2 *types.RestartWorkflowExecutionResponse, err error) {
+	if ok := h.allowDomain(ratelimitTypeUser, rp1.GetDomain()); !ok {
+		err = &types.ServiceBusyError{Message: "Too many outstanding requests to the cadence service"}
+		return
+	}
+	return h.wrapped.RestartWorkflowExecution(ctx, rp1)
+}
+
+func (h *apiHandler) ScanWorkflowExecutions(ctx context.Context, lp1 *types.ListWorkflowExecutionsRequest) (lp2 *types.ListWorkflowExecutionsResponse, err error) {
+	if ok := h.allowDomain(ratelimitTypeVisibility, lp1.GetDomain()); !ok {
```

Comment:
surprisingly, ScanWorkflowExecutions is currently classified as a "user" request.

that does seem wrong, so 👍 to this change, just calling it out for visibility.